### PR TITLE
T02- Decimal number converted into words and roman numbers.

### PR DESCRIPTION
**Pull request by Priyanka15-12**

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -1,7 +1,58 @@
...
+         } else if (num >= 20 && num < 100) {
+            int numDigit = num / 10;
+            int numRemainder = num % 10;
+            return tens[numDigit] + ((numRemainder == 0) ? "" : (" " + nums[numRemainder]));
+         } else if (num >= 100 && num < 1000) {
+            int hundredsDigit = num / 100;
+            int hunRemainder = num % 100;
+            string words = nums[hundredsDigit] + " hundred ";
+            if (hunRemainder != 0) {
+               words += "and ";
```
Comment (MetaNaveen) [inline Program.cs:28]: 720 shows "seven hundred **and** twenty"
1120 shows "one thousand one hundred **and** twenty"
but 1020 shows "one thousand twenty". shouldn't **and** be there?

File: Program.cs [hunk 2]
```diff
@@ -1,7 +1,58 @@
...
-    }
-}
\ No newline at end of file
+﻿using System;
+
+namespace Training {
+   class Program {
+      static void Main (string[] args) {
+         Console.Write ("Enter number: ");
+         int num = int.Parse (Console.ReadLine ());
```
Comment (MetaNaveen) [inline Program.cs:7]: Negative value crashes the app.
0 shows nothing.
Shouldn't you have asked for positive number?

**Review by MetaNaveen — CHANGES_REQUESTED** [review]

File: Program.cs [hunk 3]
```diff
@@ -1,7 +1,58 @@
...
+               }
+            }
+            return words;
+         } else if (num >= 1000 && num < 100000) {
+            return DecimalToWords (num / 1000) + " thousand " + (num % 1000 != 0 ? "" + DecimalToWords (num % 1000) : "");
+         } else if (num >= 100000 && num < 10000000)
+            return DecimalToWords (num / 100000) + " lakh " + (num % 100000 != 0 ? "" + DecimalToWords (num % 100000) : "");
+         return ("Out of range.");
+      }
+      static string DecimalToNumbers (int num) {
```
[the remark this answers is not in the document]
Comment (Priyanka15-12) [reply Program.cs]: "DecimalToRoman" Changed

File: Program.cs [hunk 4]
```diff
@@ -1,7 +1,58 @@
...
+         string[] romanLetters = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "XI", "V", "IV", "I" };
+         int[] numbers = { 1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1 };
+         string romanNumbers = "";
+         if (num > 0 && num <= 3999) {
```
[the remark this answers is not in the document]
Comment (Priyanka15-12) [reply Program.cs]: 5000 in Roman number is V denoted with a bar.

File: Program.cs [hunk 5]
```diff
@@ -1,7 +1,58 @@
...
+                  int hunTens = hunRemainder / 10; int hunOnes = hunRemainder % 10;
+                  words += tens[hunTens] + ((hunOnes == 0) ? "" : (" " + nums[hunOnes]));
+               }
+            }
+            return words;
+         } else if (num >= 1000 && num < 100000) {
+            return DecimalToWords (num / 1000) + " thousand " + (num % 1000 != 0 ? "" + DecimalToWords (num % 1000) : "");
+         } else if (num >= 100000 && num < 10000000)
+            return DecimalToWords (num / 100000) + " lakh " + (num % 100000 != 0 ? "" + DecimalToWords (num % 100000) : "");
+         return ("Out of range.");
```
[the remark this answers is not in the document]
Comment (Priyanka15-12) [reply Program.cs]: Not required. Changed

File: Program.cs [hunk 6]
```diff
@@ -1,7 +1,58 @@
...
+         int[] numbers = { 1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1 };
+         string romanNumbers = "";
+         if (num > 0 && num <= 3999) {
+            for (int index = 0; index < numbers.Length; index++) {
+               while (num >= numbers[index]) {
+                  romanNumbers += romanLetters[index];
+                  num -= numbers[index];
+               }
+            }
+         }return romanNumbers;
```
[the remark this answers is not in the document]
Comment (Priyanka15-12) [reply Program.cs]: Done.

File: Program.cs (hunk 1 above)
Comment (Priyanka15-12) [reply Program.cs:35]: Done.

File: Program.cs (hunk 2 above)
Comment (Priyanka15-12) [reply Program.cs]: Changed.

**Review by MetaNaveen — APPROVED** [review]

**Comment by MetaNaveen** [comment]
Working as expected.

**Final Notes:**
As per example in the doc, for tens values "-" is needed.  25 should be printed as twenty-five.
Minor formatting issues were found.

**Pull request closed without merging**